src/rdkafka_request.c

```diff
@@ -1845,3 +1845,114 @@ int rd_kafka_ProduceRequest (rd_kafka_broker_t *rkb, rd_kafka_toppar_t *rktp) {
 
         return cnt;
 }
+
+
+
+rd_kafka_resp_err_t
+rd_kafka_handle_DeleteTopics (rd_kafka_t *rk,
+                              rd_kafka_broker_t *rkb,
+                              rd_kafka_resp_err_t err,
+                              rd_kafka_buf_t *rkbuf,
+                              rd_kafka_buf_t *request,
+                              rd_kafka_topic_partition_list_t *topics) {
+        const int log_decode_errors = 1;
+        int actions;
+        int32_t TopicArrayCnt;
+        int i = 0;
+        rd_kafka_resp_err_t same_err = RD_KAFKA_RESP_ERR_NO_ERROR;
+        int same_err_cnt = 0;
+
+        if (err)
+                goto err;
+
+        rd_kafka_buf_read_i32(rkbuf, &TopicArrayCnt);
+        if (TopicArrayCnt != topics->cnt)
+                rd_kafka_buf_parse_fail(rkbuf,
+                                        "TopicArrayCnt %"PRId32" should be "
+                                        "same as %d requested topic(s)",
+                                        TopicArrayCnt, topics->cnt);
+
+        for (i = 0 ; i < (int)TopicArrayCnt ; i++) {
+                rd_kafkap_str_t topic;
+                char *topic_name;
+                rd_kafka_topic_partition_t *rktpar;
+                rd_kafka_resp_err_t ErrorCode;
+
+                rd_kafka_buf_read_str(rkbuf, &topic);
+                rd_kafka_buf_read_i16(rkbuf, &ErrorCode);
+                RD_KAFKAP_STR_DUPA(&topic_name, &topic);
+
+                rktpar = rd_kafka_topic_partition_list_find_topic(topics,
+                                                                  topic_name);
+                if (!rktpar)
+                        rd_kafka_buf_parse_fail(rkbuf,
+                                                "Non-requested topic %s "
+                                                "returned by broker",
+                                                topic_name);
+
+                rktpar->err = ErrorCode;
+
+                if (i == 0)
+                        same_err = ErrorCode;
+
+                if (same_err == ErrorCode)
+                        same_err_cnt++;
+        }
+
+        /* If all topics have the same error (or success), propagate that
+         * error as the request-level error (or success). */
+        if (same_err_cnt == TopicArrayCnt)
+                err = same_err;
```

Comment:
On this again, returning a broker_timeout error when timeout is negative seems weird (but this is more the DeleteTopicsResponse from protocol which is weird)

Also, resending a common error seems inconsitent with other method as offset handle. I would prefer error is only for general error (could not send to broker...) 
If I don't have authorization to delete topics for example, I would receive the error when all topics are valid, but a success if one inexistant topic was in the list (as all error wouldn't be the same)